.github/workflows/setupmc.yml

```diff
@@ -0,0 +1,27 @@
+            - name: Setup Minecraft
+  # You may pin to the exact commit or the version.
+  # uses: mcenv/setup-minecraft@2d092242881552b8ee463d9b4687bd38ef58a396
+  uses: mcenv/setup-minecraft@v3.3.2
+  with:
+    # Minecraft version to use
+    version: 1.12.2
+    # Whether to install Minecraft
+    install: # optional, default is true
+    # Whether to cache Minecraft
+    cache: # optional
+    # Number of retries to download Minecraft
+    retries: # optional, default is 3
+
+- uses: actions/checkout@v4
+- id: minecraft
+  uses: mcenv/setup-minecraft@v3
+  with:
+    version: "1.12.2"
+- uses: actions/setup-java@v3
+  with:
+    distribution: "microsoft"
+    java-version: ${{ fromJson(steps.minecraft.outputs.package).javaVersion.majorVersion }}
+- run: |
+    echo Running Minecraft ${{ steps.minecraft.outputs.version }}.
+    echo "eula=true" > eula.txt
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** EULA agreement handling.

Automatically agreeing to the EULA without user intervention might not be the best practice. Consider adding a step or documentation note to ensure users are aware and agree to the EULA manually.

```suggestion
    echo "eula=true" > eula.txt
    echo "Please manually agree to the Minecraft EULA located at https://account.mojang.com/documents/minecraft_eula before proceeding."
    read -p "Have you agreed to the EULA? (y/n) " agreed
    if [[ $agreed == "y" ]]; then
        echo "eula=true" > eula.txt
    else
        echo "You must agree to the EULA to run the server."
        exit 1
    fi
```